tex/chapters/MachineLearning.tex

```diff
@@ -1,3 +1,267 @@
+The model is then able to predict whether new strategies are the same as ones already implemented within Axelrod-Python.
+
+
+
+\section{Ashlock Tournament}\label{sec:ashlock_tourn}
+
+In an Ashlock tournament, all strategies entered play against a probe strategy for a range of parameters, and do not play each other.
+This is similar to how a numerical approximation of a fingerprint is constructed in Chapter \ref{cha:implementation}.
+The benefit of this approach is that much more data is collected than just the score of the strategy involved.
+Currently in Axelrod-Python, a results file for a tournament includes:
+
+\begin{itemize}
+    \item Strategy Name
+    \item Rank - Where the strategy ranked overall in the tournament
+    \item Median Score
+    \item Cooperation Rating
+    \item Wins
+    \item Initial C rate
+    \item CC rate
+    \item CD rate
+    \item DC rate
+    \item DD rate
+    \item CC to C rate
+    \item CD to C rate
+    \item DC to C rate
+    \item DD to C rate
+\end{itemize}
+
+An example of the results file for strategies that have been played in an Ashlock Tournament is given in Table \ref{tab:ash_results}.
+
+\begin{sidewaystable}[htbp!]
+\begin{tabular}{llllllllllHHH}
+\toprule
+Name &  Median\_score &  Cooperation\_rating &  Wins &  Initial\_C\_rate &  CC\_rate &  CD\_rate &  DC\_rate &  DD\_rate &  CC\_to\_C\_rate &  CD\_to\_C\_rate &  DC\_to\_C\_rate &  DD\_to\_C\_rate \\
+\midrule
+Defector & 0.59536 & 0.0000 & 22.0 & 0.0 & 0.0000 & 0.0000 & 0.4942 & 0.5058 & 0.0 & 0.0 & 0.0 & 0.0 \\
+Win-Stay Lose-Shift & 0.46868 & 0.5528 & 7.0 & 1.0 & 0.3302 & 0.2226 & 0.2264 & 0.2208 & 1.0 & 0.0 & 0.0 & 1.0 \\
+Bully & 0.46388 & 0.5024 & 11.0 & 0.0 & 0.1606 & 0.3418 & 0.3350 & 0.1626 & 0.0 & 1.0 & 0.0 & 1.0 \\
+Alternator & 0.44824 & 0.5000 &  13.0 & 1.0 & 0.2484 & 0.2516 & 0.2490 & 0.2510 & 0.0 & 0.0 & 1.0 & 1.0 \\
+ Tit For Tat & 0.43360 & 0.5144 & 0.0 & 1.0 & 0.3680 & 0.1464 & 0.1446 & 0.3410 & 1.0 & 0.0 & 1.0 & 0.0 \\
+\bottomrule
+\end{tabular}
+
+\begin{tabular}{lHHHHHHHHHlll}
+\toprule
+Name &  Median\_score &  Cooperation\_rating &  Wins &  Initial\_C\_rate &  CC\_rate &  CD\_rate &  DC\_rate &  DD\_rate &  CC\_to\_C\_rate &  CD\_to\_C\_rate &  DC\_to\_C\_rate &  DD\_to\_C\_rate \\
+\midrule
+Defector & 0.59536 & 0.0000 & 22.0 & 0.0 & 0.0000 & 0.0000 & 0.4942 & 0.5058 & 0.0 & 0.0 & 0.0 & 0.0 \\
+Win-Stay Lose-Shift & 0.46868 & 0.5528 & 7.0 & 1.0 & 0.3302 & 0.2226 & 0.2264 & 0.2208 & 1.0 & 0.0 & 0.0 & 1.0 \\
+Bully & 0.46388 & 0.5024 & 11.0 & 0.0 & 0.1606 & 0.3418 & 0.3350 & 0.1626 & 0.0 & 1.0 & 0.0 & 1.0 \\
+Alternator & 0.44824 & 0.5000 &  13.0 & 1.0 & 0.2484 & 0.2516 & 0.2490 & 0.2510 & 0.0 & 0.0 & 1.0 & 1.0 \\
+ Tit For Tat & 0.43360 & 0.5144 & 0.0 & 1.0 & 0.3680 & 0.1464 & 0.1446 & 0.3410 & 1.0 & 0.0 & 1.0 & 0.0 \\
+\bottomrule
+\end{tabular}
+\caption{Results of an Ashlock Tournament}
+\label{tab:ash_results}
+
+\bigskip
+
+\begin{tabular}{lllllllllHHHHH}
```

Comment:
Move these to where they should be. (After the point at which they're discussed).